components/SelfSubmittingSsoForm.tsx

```diff
@@ -0,0 +1,58 @@
+import { useEffect, useRef } from "react";
+import { AWS_SAML_ENDPOINT } from "../pages/api/constants";
+import { SsoDetails } from "../pages/api/types";
+
+interface Props {
+  ssoDetails: SsoDetails | undefined;
+}
+
+/**
+ * A special form designed to POST the SAML SSO details to AWS. It is also
+ * designed to auto-submit the form when the `ssoDetails` become defined. SAML
+ * logins work best when the POST as well as subsequent interactions are managed
+ * by the browser, but a user with only one configured Call Center should not
+ * have to click anything to submit this form. Thus, the form is intentionally
+ * hidden and set to auto-submit utilizing an effect hook.
+ *
+ * @param ssoDetails State variable containing the details used by the SSO form
+ */
+export default function SelfSubmittingSsoForm({ ssoDetails }: Props) {
+  // Refs seem to need `null` instead of `undefined`
+  const formRef = useRef<HTMLFormElement | null>(null);
+  const samlResponseRef = useRef<HTMLInputElement | null>(null);
+  const relayStateRef = useRef<HTMLInputElement | null>(null);
+
+  // Auto-submit the SAML form once ssoDetails are present
+  useEffect(() => {
+    if (
+      samlResponseRef.current?.value != null &&
+      samlResponseRef.current?.value?.length > 0 &&
+      relayStateRef.current?.value != null &&
+      relayStateRef.current?.value.length > 0
```

Comment:
nitpick: could this be condensed to two lines since the optionals are chained? or the optional chaining could be removed on lines 29+31?

`samlResponseRef.current?.value?.length > 0 && relayStateRef.current?.value?.length > 0`

or 

```
samlResponseRef.current?.value != null &&
      samlResponseRef.current.value.length > 0 &&
      relayStateRef.current?.value != null &&
      relayStateRef.current.value.length > 0
```